msteams-platform/sbs-create-dashboard-widget-graph-api-call-in-Teams-toolkit.yml

```diff
@@ -0,0 +1,280 @@
+
+        Teams app also supports different themes, including dark and high contrast themes.
+
+        :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/dark-light-theme.png" alt-text="Screenshot shows the contrast themes.":::
+       
+   
+- title: Add a widget to the dashboard
+  durationInMinutes: 5
+  content: |
+
+     Following are the steps to add a new widget to the dashboard:
+
+     1. [Define a data model](#define-a-data-model)
+     2. [Create a data retrieve service](#create-a-data-retrieve-service)
+     3. [Create a widget file](#create-a-widget-file)
+     4. [Add the widget to the dashboard](#add-the-widget-to-the-dashboard)
+    
+     ## Define a data model
+
+     1. In Visual Studio Code, under **EXPLORER**, go to tabs > src > models.
+     1. Create a **sampleModel.ts** file.
+     1. Use the following code and add it to the **sampleModel.ts** file:
+
+        ```
+        export interface SampleModel {
+         content: string;
+         }
+
+        ```
+
+        :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/create-sample-model.png" alt-text="Screenshot shows the creation of sampleModel file under the models folder in Visual Studio Code. ":::
+
+
+     ## Create a data retrieve service
+
+     1. In Visual Studio Code, under **EXPLORER**, go to **tabs > src > data**.
+     1. Create a **sampleData.json** file.
+     1. Use the following code in the **sampleData.json** file:
+
+        ```
+        {
+          "content": "Hello world!"
+        }
+        ```
+
+        :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/create-sample-data-json.png" alt-text="Screenshot shows the creation of sampleData.json file under the data folder in Visual Studio Code.":::
+
+     **To retrieve services**:
+
+     1. In Visual Studio Code, under **EXPLORER**, go to EXPLORER> tabs > src > data> services.
+     1. Create a **sampleService.ts** file.
+     1. Use the following code in the **sampleService.ts**:
+
+        ```
+        import { SampleModel } from "../models/sampleModel";
+        import SampleData from "../data/SampleData.json";
+
+        export const getSampleData = (): SampleModel => SampleData;
+        ```
+
+        :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/service-retrieve.png" alt-text="Screenshot shows the creation of sampleService.ts file under the service folder in Visual Studio Code.":::
+
+      > [!NOTE]
+      > You can implement a service to retrieve data from the backend service or from Microsoft Graph API.
+    
+     ## Create a widget file
+
+     1. In Visual Studio Code, go to EXPLORER> tabs> src> >views >widgets.
+     1. Create a **sampleWidget.ts** file.
+     1. Use the following code in the **sampleWidget.tsx** file:
+
+        ```
```

Comment:
For typographic issues, this code snippet looks like this. I think it might look better if indentation and syntax highlighting were added.

![image](https://user-images.githubusercontent.com/107838226/215249625-2d767ecb-5245-49c5-9c21-a1797deaa15d.png)

Reply:
Updated.